Jenkinsfile

```diff
@@ -0,0 +1,142 @@
+        )
+    }
+
+    parameters {
+        booleanParam(name: 'RELEASE', defaultValue: false, description: 'Decide whether to release')
+        choice(name: 'VERSION_BUMP', choices: ['patch', 'minor', 'major'], description: 'Which version to bump')
+    }
+    
+    stages {
+        stage('Install dependencies') {
+            steps {
+                container('ruby') {
+                    sh 'bundle install'
+                    sh 'gem install rake'
+                    sh 'wget -O semver https://raw.githubusercontent.com/fsaintjacques/semver-tool/3.3.0/src/semver && chmod +x semver'
+                }
+            }
+        }
+
+        stage('Run tests') {
+            steps {
+                container('ruby') {
+                    sh 'bundle exec rake tests'
+                }
+            }
+        }
+
+        stage('Build recog-content') {
+            steps {
+                script {
+                    OLD_VERSION = sh(script: 'git describe --abbrev=0 | cut -c 2-', returnStdout: true).trim()
+                    echo "Current version: ${OLD_VERSION}"
+
+                    VERSION = sh(script: "./semver bump ${params.VERSION_BUMP} ${OLD_VERSION}", returnStdout: true).trim()
+                    echo "New version: ${VERSION}"
+
+                    sh "zip -r recog-content-${VERSION}.zip xml"
+                }
+            }
+        }
+
+        stage('Tag / Release') {
+            when { 
+                anyOf {
+                    expression { params.RELEASE }
+                }
+            }
+
+            stages {
+                stage('Create Github tag') {
+                    steps {
+                        withCredentials([usernamePassword(credentialsId: 'github-app-key', usernameVariable: 'GH_APP', passwordVariable: 'GH_TOKEN')]) {
+                            setGitConfig()
+                            sh """git tag -a "v${VERSION}" -m "Version ${VERSION}" && git push --tags"""
+                        }
+                    }
+                }
+
+                stage('Initialise Release') {
+                    steps {
+                        withCredentials([usernamePassword(credentialsId: 'github-app-key', usernameVariable: 'GH_APP', passwordVariable: 'GH_TOKEN')]) { 
+                            sh """
+                                curl --silent --show-error \
+                                    -H "Content-Type: application/json" \
+                                    -H "Accept: application/vnd.github.v3+json" \
+                                    -H "Authorization: token \${GH_TOKEN}" \
+                                    -d '{"tag_name":"'"v${VERSION}"'", "generate_release_notes": true, "name": "'"v${VERSION} - \$(date +"%Y.%m.%d")"'"}' \
+                                    https://api.github.com/repos/rapid7/recog/releases > recog-content-releases-response.json
+                            """
+                        }
+                    }
+                }
```

Comment:
Curious whether there are any thoughts on this and the following stage `process release response`.
I think the old job used this setup to verify the release process was functioning as expected. Initialising the release and then verifying the response, before sending the actual content.

I'm not too familiar with Github releases - particularly through this endpoint. Though curious if there are any opinions on whether this may be necessary? Or perhaps it could be simplified down into a single request.

Reply:
@dtwaddle2-r7 That is correct. The `Process release response` stage is to verify there were not errors in the previous `Initialise Release` stage.

See the comment on 82-83:
```
// Note: Inspect the response for error message since curl exits with code 0
// for the "Validation failed" response with status code 422 Unprocessable Entity
```

Reply:
Please note that it is also performing more work than verifying the release. It processes that response to get values that are used in other stages such as `UPLOAD_URL`.

Reply:
Ah, thank you for clarifying. I initially thought it may be possible to combine this initial release with the one used to submit the zip content. That perhaps this approach was done for additional validation / debugging.
Though yeah if it has to be separated, then this makes sense. Thanks.

Reply:
You might be able to combine the work with other stages. Thinking back I may have broken it down to better determine the cause of a failure. If curl happened to exit with a non-zero then the `Initialise Release` step would fail and you would know that was due to a network or auth issue. If the job failed in `Process release response` it would based on the API response. See: [HTTP response status codes for "Create a release"](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release--status-codes).